index.js

```diff
@@ -548,20 +605,72 @@ class KeyringController extends EventEmitter {
    * @param {string} password - The keyring controller password.
    * @returns {Promise<Array<Keyring>>} The keyrings.
    */
-  async unlockKeyrings(password) {
+  async unlockKeyrings(password, encryptedKey) {
     const encryptedVault = this.store.getState().vault;
     if (!encryptedVault) {
       throw new Error('Cannot unlock without a previous vault.');
     }
 
     await this.clearKeyrings();
-    const vault = await this.encryptor.decrypt(password, encryptedVault);
-    this.password = password;
+
+    // MV3: If the separator string is in the vault string, the user has already migrated
+    // from the previous password-only model
+    let vault = null;
+    if (encryptedVault?.includes?.(VAULT_SEPARATOR)) {
+      const { salt, vault: vaultOnly } = this.parseVault(encryptedVault);
```

Comment:
(nit)
since this codebase is OO style more than it is FP, maybe a simple Vault class with a constructor that can create the vault object by parsing the string or getting both chunks as input would look nicer. Something like the global URL() which can be used to parse URLs or create them from pieces.

It'd expose public fields:
- salt
- vault
- serialized (the string resulting from concatenation or passed in for parsing)
- isSalted (to use instead of `encryptedVault?.includes?.(VAULT_SEPARATOR)` here)

SEPARATOR would not be known outside the Vault class